tests/unit/test_config.py

```diff
@@ -366,13 +366,20 @@ def test_process_icp(
 
 
 @pytest.mark.parametrize(
-    'startcp, expected',
-    [('20210120T1700+0530', '20210120T1700+0530'),
-     ('now', '20050102T0615+0530'),
-     (None, '18990501T0000+0530')]
+    'startcp, starttask, expected',
+    [
+        ('20210120T1700+0530', None, '20210120T1700+0530'),
+        ('now', None, '20050102T0615+0530'),
+        (None, None, '18990501T0000+0530'),
+        (
+            None,
+            ['foo.20090802T0615+0530', 'bar.20090802T0515+0530'],
+            '20090802T0515+0530'
+        )
+    ]
```

Comment:
I think there should be a test case where both `startcp` and `starttask` are specified (with the former winning, according to how you've implemented it)

Reply:
I've made the two options mutually exclusive (which they should be, as noted by @dpmatthews in Element chat recently) and updated the unit test for that.